### PR TITLE
Issue #73: use full checkstyle run in UTs

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #73: use full checkstyle run in UTs

this PR is based on https://github.com/checkstyle/patch-filters/pull/84 and https://github.com/checkstyle/patch-filters/pull/87, so it is better to review https://github.com/checkstyle/patch-filters/pull/84 and https://github.com/checkstyle/patch-filters/pull/87 first, and then review this PR.

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java [hunk 1]
```diff
@@ -180,36 +180,242 @@ private void shouldRejectLine(SuppressionPatchFilter filter, int lineNo, String
         return createSuppressionPatchFilter(fileName, "changed");
     }
 
-    @Ignore
     @Test
+    @Ignore("wait until testByConfig can process a dir contains no more than one file")
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java:184]: please make it like `until http://...link.. when testByconfig .....`

the same for other tests.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/FileTabCharacter/newline/zeroSizeConfig.xml [hunk 2]
```diff
@@ -6,4 +6,8 @@
   <module name="FileTabCharacter">
     <property name="eachLine" value="true"/>
   </module>
+  <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchFilter">
+    <property name="file" value="src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/FileTabCharacter/zeroSize.patch" />
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/FileTabCharacter/newline/zeroSizeConfig.xml:10]: rename file to `zeroContext.patch`

https://en.wikipedia.org/wiki/Diff#Context_format

do same for other patch files.
please rename all java variables to keep `Context` in name.

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java (hunk 1 above)
Comment (HuGanghui) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/FileTabCharacter/newline/zeroSizeConfig.xml (hunk 2 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/strategy/FileTabCharacter/newline/zeroSizeConfig.xml]: done

**Pull request merged**